src/server/auth/server/api_server.go

```diff
@@ -1175,6 +1195,252 @@ func (a *apiServer) RevokeAuthToken(ctx context.Context, req *authclient.RevokeA
 	return &authclient.RevokeAuthTokenResponse{}, nil
 }
 
+func (a *apiServer) SetGroupsForUser(ctx context.Context, req *authclient.SetGroupsForUserRequest) (resp *authclient.SetGroupsForUserResponse, retErr error) {
+	a.LogReq(req)
+	defer func(start time.Time) { a.LogResp(req, resp, retErr, time.Since(start)) }(time.Now())
+	if !a.isActivated() {
+		return nil, authclient.ErrNotActivated
+	}
+
+	callerInfo, err := a.getAuthenticatedUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if !a.isAdmin(callerInfo.Subject) {
+		return nil, &authclient.ErrNotAuthorized{
+			Subject: callerInfo.Subject,
+			AdminOp: "SetGroupsForUser",
+		}
+	}
+
+	username, err := lenientCanonicalizeSubject(ctx, req.Username)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := col.NewSTM(ctx, a.etcdClient, func(stm col.STM) error {
+		members := a.members.ReadWrite(stm)
+
+		// Get groups to remove/add user from/to
+		var removeGroups authclient.Groups
+		addGroups := addToSet(nil, req.Groups...)
+		if err := members.Get(username, &removeGroups); err == nil {
+			for _, group := range req.Groups {
+				if _, ok := removeGroups.Groups[group]; ok {
```

Comment:
I think that every value in this map is `true` right? That means you could make this
```
if removeGroups.Groups[group] {
...
}
```
if you wanted

Reply:
Yeah, that is true. I tend to stick to the comma ok idiom for checking key existence, but since this is being used as a set it might be better to check that way.